Improvement/zenko 4982 extra env vars secret specification

benzekrimaha opened the pull request:
Issue: ZENKO-4982

bert-e commented:
# Hello benzekrimaha,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4982 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.11.4`


Please check the `Fix Version/s` of ZENKO-4982, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented on solution-base/mongodb/charts/mongodb-sharded/values.yaml:
```diff
@@ -509,13 +509,28 @@ configsvr:
   ##  - name: KIBANA_ELASTICSEARCH_URL
   ##    value: test
   ##
```
we don't use `values.yaml` to specify the parameters, we set them on the parameters to helm template, in `build.sh` : any reason not to do this here?

francoisferrand commented on solution-base/mongodb/charts/mongodb-sharded/values.yaml:
```diff
@@ -509,13 +509,28 @@ configsvr:
   ## @param configsvr.extraEnvVarsSecret Name of a Secret containing extra env vars
   ##
-  extraEnvVarsSecret: ""
+  extraEnvVarsSecret: "mongodb-db-creds"
```
setting this will add separate entries:
```
envFrom:
  - secretRef:
    name: "mongodb-db-creds"
```

i.e. it will create extra env variables...

francoisferrand replied:
still not addressed: `--set shardsvr.dataNode.extraEnvVarsSecret=${MONGODB_NAME}-db-creds` and `--set configsvr.extraEnvVarsSecret=${MONGODB_NAME}-db-creds` will generate an `envFrom` directive, which will inject extra secrets

benzekrimaha replied:
Indeed not needed done here : https://github.com/scality/Zenko/pull/2210/commits/999e275d35a9aa789e9467c1890b15ab8177580b

francoisferrand commented on solution-base/build.sh:
```diff
@@ -121,7 +121,37 @@ function render_mongodb_sharded_yamls()
+        --set 'configsvr.extraEnvVars[2].valueFrom.secretKeyRef.name=mongodb-db-creds' \
+        --set 'configsvr.extraEnvVars[2].valueFrom.secretKeyRef.key=mongodb-password' \
+        --set configsvr.extraEnvVarsSecret=${MONGODB_NAME}-db-creds \
+        --set 'shardsvr.arbiter.extraEnvVars[0].name=MONGODB_APP_USERNAME' \
+        --set 'shardsvr.arbiter.extraEnvVars[0].valueFrom.secretKeyRef.name=mongodb-db-creds' \
+        --set 'shardsvr.arbiter.extraEnvVars[0].valueFrom.secretKeyRef.key=mongodb-username' \
+        --set 'shardsvr.arbiter.extraEnvVars[1].name=MONGODB_APP_DATABASE' \
+        --set 'shardsvr.arbiter.extraEnvVars[1].valueFrom.secretKeyRef.name=mongodb-db-creds' \
+        --set 'shardsvr.arbiter.extraEnvVars[1].valueFrom.secretKeyRef.key=mongodb-database' \
+        --set 'shardsvr.arbiter.extraEnvVars[2].name=MONGODB_APP_PASSWORD' \
+        --set 'shardsvr.arbiter.extraEnvVars[2].valueFrom.secretKeyRef.name=mongodb-db-creds' \
+        --set 'shardsvr.arbiter.extraEnvVars[2].valueFrom.secretKeyRef.key=mongodb-password' \
+        --set shardsvr.arbiter.extraEnvVarsSecret=${MONGODB_NAME}-db-creds \
```
arbiter not needed, we don't have arbiters (and the chart is pre-rendered during build)

benzekrimaha committed:
putting secrets in build.sh instead of using patch on charts

This commit aims to simplify the process of managing secrets in the build process by
removing the patching of charts and instead directly including the secrets in the build.sh script.
Issue: ZENKO-4982

benzekrimaha commented:
/approve

bert-e commented:
# Build failed



The [build](https://github.com/scality/Zenko/actions?query=branch%3Aimprovement/ZENKO-4982-extraEnvVarsSecret-specification) for [commit](https://github.com/scality/Zenko/commit/a3f5dbf36d9925bd67675f78a96fd885cfaa87e9) did not succeed in branch [improvement/ZENKO-4982-extraEnvVarsSecret-specification](https://github.com/scality/zenko/compare/development/2.11...improvement/ZENKO-4982-extraEnvVarsSecret-specification)



*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.11`



The following branches have **NOT** changed:

* `development/2.10`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue ZENKO-4982.

Goodbye benzekrimaha.




*The following options are set:* **approve**
